src/parser/ast/expression.rs

```diff
@@ -0,0 +1,215 @@
+use std::fmt::Debug;
+
+use super::DebugSymRef;
+
+#[derive(Clone)]
+pub enum BinaryOperator {
+    Sum,
+    Sub,
+    Mul,
+    Pow,
+    Eq,
+    NEq,
+    And,
+    Or,
+}
+
+impl BinaryOperator {
+    pub fn is_logic(&self) -> bool {
+        use BinaryOperator::*;
+        match self {
+            Eq => true,
+            NEq => true,
+            And => true,
+            Or => true,
+            Sum => false,
+            _ => false,
+        }
+    }
+
+    pub fn is_arith(&self) -> bool {
+        !self.is_logic()
+    }
+}
+
+impl Debug for BinaryOperator {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Self::Sum => write!(f, "+"),
+            Self::Sub => write!(f, "-"),
+            Self::Mul => write!(f, "*"),
+            Self::Pow => write!(f, "^"),
+            Self::Eq => write!(f, "=="),
+            Self::NEq => write!(f, "!="),
+            Self::And => write!(f, "&&"),
+            Self::Or => write!(f, "||"),
+        }
+    }
+}
+
+impl From<String> for BinaryOperator {
+    fn from(value: String) -> Self {
+        use BinaryOperator::*;
+        match value.as_str() {
+            "+" => Sum,
+            "*" => Mul,
+            "^" => Pow,
+            "==" => Eq,
+            "!=" => NEq,
+            "&&" => Or,
+            "||" => And,
+            &_ => unimplemented!(),
```

Comment:
unimplemented? or unreachable?  Will this be an Error in the future?

Reply:
Sorry for the intromission here, but actually I think unimplemented can make sense because it means that anything else besides what was defined previously is not implemented, right?

Reply:
Yes, it is a way to throw an error. This is called by the parser and as the user does not have a way to include new operators calling this with a different symbol would be a bug, not an error that the user could trigger, so I would not implement an actual error reporting system for this.

Reply:
Yes, it should be unreachable

Reply:
Done